src/model/scope/scope.ts

```diff
@@ -0,0 +1,107 @@
+/* istanbul ignore file */
+
+/*
+ * This flag is to ignore BDD testing for model
+ * which will be addressed in the future in
+ * ticket #354
+ * /
+
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop
+ files are distributed onan 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+import Knex from 'knex'
+import { NotFoundError, IllegalOperationError } from '../errors'
+
+/*
+* Interface for Scope resource type
+*/
+export interface Scope {
+  id?: number;
+  consentId: string;
+  action: string;
+  accountId: string;
+}
+
+/*
+ * Class to abstract Scope DB operations
+ */
+export class ScopeDB {
+  // Knex instance
+  private Db: Knex
+
+  public constructor (dbInstance: Knex) {
+    this.Db = dbInstance
+  }
+
+  // Add a single Scope or an array of Scopes
+  public async insert (scopes: Scope | Scope[]): Promise<boolean> {
+    if (Array.isArray(scopes) && scopes.length === 0) {
+      throw new IllegalOperationError('Scope')
+    }
+
+    // Returns [0] for MySQL-Knex and [Row Count] for SQLite-Knex
+    await this
+      .Db<Scope>('Scope')
+      .insert(scopes)
+
+    return true
+  }
+
+  // Retrieve Scopes by Consent ID
+  public async retrieveAll (consentId: string): Promise<Scope[]> {
+    const scopes: Scope[] = await this
+      .Db<Scope>('Scope')
+      .select('*')
+      .where({ consentId })
+
+    // Not dinguishing between a Consent that exists
+    // with 0 scopes and a Consent that does not exist
+    if (scopes.length === 0) {
+      throw new NotFoundError('Consent\'s Scopes', consentId)
+    }
+
+    return scopes
+  }
+
+  // Delete Scopes by Consent ID
+  public async deleteAll (consentId: string): Promise<number> {
+    const deleteCount: number = await this
```

Comment:
what if `consentId` is empty string?

Reply:
if we decided to do input validation let be consistent

Reply:
There is no input validation in the models @eoln . The only place there is validation is for Empty array in insert because it returns a "SQL MISUSE: not an error" exception which is also dependent on the DB. So I added it to standardize it.

Reply:
If you pass in garbage, I'm not sure if it's the model's job to deal with your garbage. The Handler layer should look at this and decide to reject bad input. If the result of an empty string was a bunch of "technically correct" but not quite correct data returned, then I'd say we should defend against that. But for an empty string you'll just get back zero results, so that should be fine.